src/live-preview/storage-helper.js

```diff
@@ -0,0 +1,43 @@
+// This is a helper class to help store key-value data inside
+// an individual Storage object's key. 
+// It simply performs the JSON parsing and stringifying steps
+
+export class Storage {
+    storage;
+    name;
+
+    /**
+     * @param {string} name 
+     * @param {Storage.prototype} storage 
+     */
+    constructor(storage, name) {
+        if (!storage?.__proto__ === Storage.prototype) {
+            throw new Error("storage should implment Storage")
```

Comment:
could you verify this statement looks a bit off

Reply:
Ensures that the storage parameter is implementing the [Web Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Storage). So the class can be used with local or session or any Storage object.